video/out/hwdec/dmabuf_interop_gl.c

```diff
@@ -114,44 +103,95 @@ static bool vaapi_gl_mapper_init(struct ra_hwdec_mapper *mapper,
     return true;
 }
 
+static void gl_delete_textures(const struct ra_hwdec_mapper *mapper)
+{
+    struct dmabuf_interop_priv *p_mapper = mapper->priv;
+    struct vaapi_gl_mapper_priv *p = p_mapper->interop_mapper_priv;
+
+    GL *gl = ra_gl_get(mapper->ra);
+    gl->DeleteTextures(4, p->gl_textures);
+    for (int n = 0; n < 4; n++) {
```

Comment:
This `4` is pretty magic, could we avoid that?

Reply:
It's a static array so we can use an array count macro.

Reply:
This constant is scattered around the code:
<pre>
video/out/hwdec/dmabuf_interop_gl.c
55:    GLuint gl_textures[4];
56:    EGLImageKHR images[4];
58:    const struct ra_format *planes[4];
75:    gl->GenTextures(4, p->gl_textures);
112:    gl->DeleteTextures(4, p->gl_textures);
113:    for (int n = 0; n < 4; n++) {
327:    for (int n = 0; n < 4; n++) {

video/out/hwdec/dmabuf_interop.h
41:    struct ra_tex *tex[4];

video/out/hwdec/dmabuf_interop_pl.c
113:    for (int n = 0; n < 4; n++)
</pre>
`AV_DRM_MAX_PLANES` would be suitable, but would break if ever changed because our RA has an identical limit (with no define):
<pre>
video/out/gpu/ra.h
104:    int num_components;     // component count, 0 if not applicable, max. 4
105:    int component_size[4];  // in bits, all entries 0 if not applicable
106:    int component_depth[4]; // bits in use for each component, 0 if not applicable
459:    void (*clear)(struct ra *ra, struct ra_tex *dst, float color[4],
530:    const struct ra_format *planes[4];
549:    uint8_t components[4][4];
</pre>

Reply:
I know mpv organization is not paying for fixing bugs like that, unlike ffmpeg. But maybe it would be still beneficial to do such a fix?

Reply:
if someone has lots of time some day they can go through all the mpv code and replace 4 with something like `MAX_PLANES` (it's not just hwdec, basically everywhere)